llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -849,7 +849,8 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
     AddPromotedToType(Op, MVT::bf16, MVT::f32);
   }
   for (const auto &Op : {ISD::FABS}) {
-    setOperationAction(Op, MVT::f16, Promote);
+    // Expand instead of Promote to clear sign bit by bitcasting to i16
+    setOperationAction(Op, MVT::f16, Expand);
```

Comment:
Would operations other than FABS benefit from that, too?  E.g. `fneg` or `fcopysign`?

Reply:
that's a valid point.

For `NVPTX`, we have for `f16` (so no change required)

* `FNEG` Legal / Expand
* `FCOPYSIGN` Expand

For `X86-64`, we have for `f16`

* `FNEG` Promote
* `FCOPYSIGN` Expand